src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/ConvertToLibraryImportFixer.cs

```diff
@@ -602,11 +605,39 @@ private static SyntaxNode SortDllImportAttributeArguments(AttributeSyntax attrib
                 return false;
             }
 
+            // By default we always prefer collection expressions for C# 12 and above
+            bool useCollectionExpression = ((CSharpParseOptions)documentSyntaxTree.Options).LanguageVersion >= LanguageVersion.CSharp12;
```

Comment:
Can you extract all of the "should we use collection expressions" logic into a separate method?